larpandoracontent/LArHelpers/LArFormattingHelper.h

```diff
@@ -0,0 +1,222 @@
+/**
+ *  @file   larpandoracontent/LArHelpers/LArFormatting.h
+ *
+ *  @brief  Header file for the formatting helper class
+ *
+ *  $Log: $
+ */
+#ifndef LAR_FORMATTING_HELPER_H
+#define LAR_FORMATTING_HELPER_H 1
+
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <vector>
+
+namespace lar_content
+{
+
+/**
+ *  @brief  LArFormattingHelper class
+ */
+class LArFormattingHelper
+{
+public:
+
+    enum Colour : int;
+    enum Style : int;
+
+    /**
+     *  @brief  Set the format style (to standard output stream)
+     *
+     *  @param  style the style of choice
+     */
+    static void SetStyle(const Style &style, std::ostream &stream=std::cout);
+    
+    /**
+     *  @brief  Set the text colour (of standard output stream)
+     *
+     *  @param  colour the colour of choice
+     */
+    static void SetColour(const Colour &colour, std::ostream &stream=std::cout);
+    
+    /**
+     *  @brief  Reset the style of the standard output stream
+     */
+    static void ResetStyle(std::ostream &stream=std::cout);
+
+    /**
+     *  @brief  Reset the text colour of the standard output stream
+     */
+    static void ResetColour(std::ostream &stream=std::cout);
+
+    /**
+     *  @brief  Reset the formatting and text colour of the standard output stream
+     */
+    static void Reset(std::ostream &stream=std::cout);
+
+    /**
+     *  @brief  Print a formatting character to the standard output stream
+     *
+     *  @param  code the formatting code to output
+     */
+    static void PrintFormatCharacter(const unsigned int &code, std::ostream &stream=std::cout);
+
+    /**
+     *  @brief  Get a formatting character
+     *
+     *  @param  code the formatting code to output
+     */
+    static std::string GetFormatCharacter(const unsigned int &code);
+
+    /**
+     *  @brief  Print a header line of a given width
+     *
+     *  @param  title the title of the header
+     *  @param  width the width of the header line
+     */
+    static void PrintHeader(const std::string &title = "", const unsigned int &width = 140);
+
+    /**
+     *  @brief  Print a horizontal rule
+     *
+     *  @param  width the width of the rule line
+     */
+    static void PrintRule(const unsigned int &width = 140);
+
+    enum Style : int
```

Comment:
doxygen: @brief, etc.

Where did all the magic numbers come from and can they be accessed via a nice header file somewhere?

Reply:
I can't actually find any standard headers, and I think that's because the definitions are slightly different across platforms. From what I understand this seems to be mainly windows vs. non-windows, which isn't a problem for us! But it would be good if we could test on mac.

I did find a nifty snippet that we could use to check if a users terminal supports colours, by reading the `TERM` environment variable. 
```c++
bool supportsColor()
{
    if(const char *env_p = std::getenv("TERM")) {
        const char *const terms[] = {
            "xterm", "xterm-256", "xterm-256color", "vt100",
            "color", "ansi",      "cygwin",         "linux"};
        // Change 1
        for(auto const term: terms) {
            if(std::strcmp(env_p, term) == 0) return true;
        }
    }
    return false;
}
```
This bool would only need to used in `GetFormatCharacter`, which could always return `""` if colour isn't supported. What do you think?